weblate/trans/forms.py

```diff
@@ -2498,7 +2525,13 @@ def get_new_unit_form(translation, user, data=None, initial=None):
             translation, user, data=data, initial=initial
         )
     if translation.is_source:
-        return NewBilingualSourceUnitForm(translation, user, data=data, initial=initial)
+        return NewBilingualSourceUnitForm(
+            translation,
+            user,
+            data=data,
+            initial=initial,
+            is_source_plural=is_source_plural,
+        )
     return NewBilingualUnitForm(translation, user, data=data, initial=initial)
```

Comment:
At least this also needs the logic (when adding string to bilingual translation), it still lacks the plural fields:

![image](https://github.com/WeblateOrg/weblate/assets/212189/7f55efbb-0945-4557-ab43-de05fe462a60)

I'm not sure about the glossary forms above. Technically, there is no reason not to use plurals, but the default glossary format doesn't support it.

Reply:
Figured this out, both `source` and `target` in the `NewBilingualUnitForm` must receive the value of `is_source_plural`.